cql3/functions/vector_search_fcts.cc

```diff
@@ -0,0 +1,61 @@
+/*
+ * Modified by ScyllaDB
+ *
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (LicenseRef-ScyllaDB-Source-Available-1.0 and Apache-2.0)
+ */
+
+#include "vector_search_fcts.hh"
+#include "cql3/query_options.hh"
+
+namespace cql3 {
+namespace functions {
+namespace {
+
+static size_t find_matching_key_index(const expr::evaluation_inputs& inputs, const std::vector<cql3::statements::primary_key>& keys) {
+    for (size_t i = 0; i < keys.size(); ++i) {
```

Comment:
I don't understand what I'm seeing here. What is the "keys" vector - how big is it? Is it the entire O(N) result list? So this function is O(N)? And we call it for every result? If so, this code has quadratic complexity, which is pretty bad.
Or am I misunderstanding what this function does? Maybe it's a sign it needs a comment.